cryostatio/cryostat-web
# fix(notifications): handle overflow

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Fixes: #757

## Description of the change:
This change adds overflow handling to the notification alert group and notification drawer.

https://www.patternfly.org/v4/components/alert-group#toast-alert-group-with-overflow-capture

## Motivation for the change:
This reduces visual clutter on the screen and keeps the UI from feeling overfilled and chaotic when many actions are happening within a short timeframe. It is also easier for the user to dismiss all notifications when there are very many since the notification drawer can be summoned in a second way (clicking the overflow message), and summoning the drawer also hides the toast boxes.

## How to manually test:
There is a new temporary plus button on the top bar to help testing. This is to be removed before merge. Clicking it adds a new notification immediately.

Generate notifications with the temporary button or by any other means and observe the Overflow message that appears when there are too many notifications. Go to the Settings view and try changing the visible notifications count to various values and repeat.
